internal/shared/git/git.go

```diff
@@ -5,91 +5,121 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/go-git/go-git/v5/config"
+
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing/transport"
 	gitHttp "github.com/go-git/go-git/v5/plumbing/transport/http"
 	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
 
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
 	controllerConfig "github.com/numaproj-labs/numaplane/internal/controller/config"
 	"github.com/numaproj-labs/numaplane/internal/shared/kubernetes"
-	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// GetAuthMethod returns an authMethod  for both cloning and fetching from a repo with HTTP, SSH, or TLS credentials from Kubernetes secrets.
+func GetAuthMethod(ctx context.Context, repoCred *controllerConfig.RepoCredential, kubeClient k8sClient.Client, repoUrl string) (transport.AuthMethod, bool, error) {
+	scheme, err := GetURLScheme(repoUrl)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to parse URL scheme: %w", err)
+	}
+
+	var auth transport.AuthMethod
+	var insecureSkipTLS bool
+	if repoCred != nil {
+		// Configure TLS if applicable
+		if repoCred.TLS != nil {
+			insecureSkipTLS = repoCred.TLS.InsecureSkipVerify
+		}
+
+		switch scheme {
+		case "http", "https":
+			if cred := repoCred.HTTPCredential; cred != nil {
```

Comment:
I think this will panic if somebody's repo configures http as their repository path scheme and has this URL in their ConfigMap but doesn't include the HTTPCredential in it, right? (same for ssh)

Reply:
If the repository path scheme is http, and there's no matching credential, we should try to connect with no credential.

Reply:
I have covered that test case - https://github.com/numaproj-labs/numaplane/blob/feat/fetchAuth/internal/shared/git/git_test.go#L205

Reply:
Yeah in that case auth will be nil ,the clone options wont have the auth which will be nil -I have covered that oart 
 in this test case -https://github.com/numaproj-labs/numaplane/blob/feat/fetchAuth/internal/shared/git/git_test.go#L205

Reply:
sorry, my mistake! I didn't read the `cred != nil` part :(